crates/typst-pdf/src/gradient.rs

```diff
@@ -146,8 +146,9 @@ pub fn write_gradients(
                         .bits_per_flag(8)
                         .shading_type(StreamShadingType::CoonsPatch)
                         .decode([
-                            0.0, 1.0, 0.0, 1.0, range[0], range[1], range[2], range[3],
-                            range[4], range[5],
+                            0.0, 1.0, 0.0, 1.0,
```

Comment:
This should therefore also be "programmable" otherwise it will waste file size

Reply:
Good point! Tentative solution with the parameter of `decode` having (if I'm not mistaken) the right size for each color space:

https://github.com/typst/typst/pull/4568/files#diff-6a20014eeaaa0cd7d097c8ead96193dfc6ed65aea4ed7ff2de0b5584b92dee78L143-R175

Not sure this is a good way to do it as it introduces quite a bit of code duplication. Maybe there is a way to move the `match` inside `decode`? I'll think more about it.